src/pudl/transform/eia930.py

```diff
@@ -0,0 +1,224 @@
+        )
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "subregion_code_eia": pd.CategoricalDtype(),
+            }
+        )
+        .loc[
+            :,
+            [
+                "report_datetime_local",
+                "report_datetime_utc",
+                "balancing_authority_code_eia",
+                "subregion_code_eia",
+                "demand_reported_mw",
+            ],
+        ]
+    )
+    assn = (
+        demand.groupby("balancing_authority_code_eia")["subregion_code_eia"]
+        .unique()
+        .explode()
+        .to_frame()
+        .reset_index()
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "subregion_code_eia": pd.CategoricalDtype(),
+            }
+        )
+    )
+    return (
+        Output(value=demand, output_name="core_eia930__hourly_subregion_demand"),
+        Output(
+            value=assn, output_name="core_eia930__assn_balancing_authority_subregion"
+        ),
+    )
+
+
+@multi_asset(
+    outs={
+        "core_eia930__hourly_balancing_authority_interchange": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "core_eia930__assn_balancing_authority_region": AssetOut(
+            io_manager_key="pudl_io_manager"
+        ),
+    },
+    compute_kind="pandas",
+)
+def core_eia930__hourly_balancing_authority_interchange_assets(
+    raw_eia930__interchange: pd.DataFrame,
+):
+    """Produce a normalized table of hourly interchange by balancing authority.
+
+    * region_code_eia and adjacent_region_code_eia are from the same set of values, but
+      adjacent_region_code_eia also includes "CAN" and "MEX" because foreign countries
+      can be adjacent, but are not reporting directly.
+    * similarly adjacent_balancing_authority_and balancing_authority_code come from the
+      same pool of values, but the adjancent_balancing_authority_code_eia includes some
+      codes that hail from Canada and Mexico.
+    * There's an implied set of associations between the regions and the balancing
+      authorities.
+    * Need to check for consistency with BA codes mentioned here and elsewhere in
+      PUDL (plants table, EIA861 BA table).
+
+    """
+    interchange = raw_eia930__interchange.astype(
+        {
+            "balancing_authority_code_eia": pd.CategoricalDtype(),
+            "adjacent_balancing_authority_code_eia": pd.CategoricalDtype(),
+            "region_code_eia": pd.CategoricalDtype(),
+            "adjacent_region_code_eia": pd.CategoricalDtype(),
+        }
+    ).loc[
+        :,
+        [
+            "report_datetime_local",
+            "report_datetime_utc",
+            "balancing_authority_code_eia",
+            "adjacent_balancing_authority_code_eia",
+            "region_code_eia",
+            "adjacent_region_code_eia",
```

Comment:
The region codes are technically superfluous because they're implied by the BA codes, if we wanted a more normalized version of this table.